### ISSUE TITLE
Unable to retry got.stream

### ISSUE THREAD

**Issue by ultimate-tester** [sindresorhus/got#1452]
#### What would you like to discuss?
When piping a stream from got, I get a consistent ECONNRESET error after roughly 1 minute of piping (I'm piping several gigabytes). 
Unfortunately I'm unable to implement the retry approach written in the README (basically recall the function manually as long as the "retry" event is called. ). The "retry" event never triggers, it fails immediately. How should I approach retrying the got stream piping?

#### Checklist

- [X] I have read the documentation.

**Comment by szmarczak** [comment]
Can you post some code to reproduce the issue please?

**Comment by ultimate-tester** [comment]
> 
> 
> Can you post some code to reproduce the issue please?

Thanks for your reply, I've updated my post to include the code.

**Comment by mnsrulz** [comment]
@szmarczak 
is it possible to resume the input stream from where it left off? I am assuming we need to implement something ourself which tracks how many bytes have been already piped and resuming from there but just confirming...

**Comment by szmarczak** [comment]
> is it possible to resume the input stream from where it left off?

If you mean uploading data, then no such thing exists in streams at all. You'd have to implement this on your own. Basically reupload.

**Comment by mnsrulz** [comment]
yeah, a reupload with smart detection which allows starting from where it left off. So in google drive, when an error occurred while uploading a stream, google exposes an API that can tell how much data you already sent for a given upload. So, if I am uploading 1gb file and the process errored out at 300mb then the input stream can start from 301mb. 

so essentially, I am looking for a readable stream that is piped from the got input stream and if that input stream errored out I can start pumping from the exact position where it left off. sounds doable to me but requires a good knowledge of streams.
